src/Middleware/src/Headstart.API/Commands/StorefrontCommand.cs

```
@@ -0,0 +1,125 @@
+    {
+        private readonly AppSettings _settings;
+        private readonly IOrderCloudIntegrationsBlobService _blob;
+        private readonly IOrderCloudClient _oc;
+
+
+        public StorefrontCommand(IOrderCloudIntegrationsBlobService blob, AppSettings settings, IOrderCloudClient oc)
+        {
+            _settings = settings;
+            _blob = blob;
+            _oc = oc;
+        }
+
+        private readonly string storefrontPlaceholder = "<PLACEHOLDER>";
+        private readonly string envPlaceholder = "\"#{environmentConfig}\"";
+        private readonly string originalMainFileName = "main-original.js";
+        private bool isMainFile(string fileName) => fileName.StartsWith("main.") && fileName.EndsWith(".js");
+        
+
+        public async Task DeployBuyerSite(ApiClient apiClient = null)
+        {
+            //First create the api client
+            //var client = await _oc.ApiClients.CreateAsync(apiClient);
+
+
+            //1. Create the $web container (not sure if this is necessary. If they enable static website hosting this should already be there)
+            await _blob.CreateContainerAsync("$web", false);
+
+            //2. Get files of deployed buyer app so we can upload them to our contianer (within a subfolder)
+            var files = await _blob.GetBlobFiles("buyerweb");
+            var tasks = new List<Task>();
+
+            var storefrontName = apiClient?.xp?.StorefrontName;
+            var incrementorPrefix = apiClient?.xp?.IncrementorPrefix;
+            var appName = apiClient.AppName;
+            var mainFileName = "";
+
+            foreach (var file in files)
+            {
+                var fileName = file.Uri.ToString().Split("buyerweb/")[1];
+                if(isMainFile(fileName))
+                {
+                    mainFileName = fileName; // need to save the mainFileName to a variable so we know which file to replace with the new app configs
```

We probably want to throw an error if mainFilename isn't found

---

Adding error.